homeassistant/components/rainmachine/__init__.py

```diff
@@ -109,6 +130,15 @@
     }
 )
 
+SERVICE_PUSH_FLOW_METER_DATA_SCHEMA = SERVICE_SCHEMA.extend(
+    {
+        vol.Required(CONF_VALUE): cv.positive_float,
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): vol.All(
+            cv.string, valid_flow_meter_units
```

Comment:
This should be sufficient:

```suggestion
            cv.string, vol.In(CV_FLOW_METER_VALID_UNITS)
```